compiler/optimizations/forallOptimizations.cpp

```diff
@@ -1270,24 +1290,25 @@ static Symbol *generateStaticCheckForAccess(ALACandidate& candidate,
   if (staticCheckSymMap.count(baseSym) == 0) {
     SET_LINENO(forall);
 
-    VarSymbol *checkSym = new VarSymbol("chpl__staticAutoLocalCheckSym");
+    VarSymbol *checkSym = new VarSymbol("chpl__ala_staticCheckSym");
     checkSym->addFlag(FLAG_PARAM);
     // mark it with FLAG_TEMP to prevent the normalizer from adding
     // PRIM_END_OF_STATEMENT in the wrong places for loops.
     checkSym->addFlag(FLAG_TEMP);
     staticCheckSymMap[baseSym] = checkSym;
 
-    CallExpr *checkCall = new CallExpr("chpl__staticAutoLocalCheck");
+    CallExpr *checkCall = new CallExpr("chpl__ala_staticCheck");
     checkCall->insertAtTail(baseSym);
 
-    if (optInfo.iterSym[iterandIdx] != NULL) {
-      checkCall->insertAtTail(new SymExpr(optInfo.iterSym[iterandIdx]));
+    if (Expr* e = optInfo.getLoopDomainExpr()) {
+      checkCall->insertAtTail(e);
     }
-    else if (optInfo.dotDomIterExpr[iterandIdx] != NULL) {
-      checkCall->insertAtTail(optInfo.dotDomIterExpr[iterandIdx]->copy());
+    else {
+      INT_FATAL("optInfo didn't have enough information");
```

Comment:
This looks like a bug report waiting to happen. Is there a way to gracefully bail out of the optimization?

Reply:
I don't disagree. But I don't see an easy way of achieving what you want. At this point, we are a bit too deep in transformation and the change required for that is more than I prefer for this PR.

Talking a bit more about that `INT_FATAL`, note that similar ones existed since the first iterations of the optimization which dates back several releases. To me, this is not much different than `INT_ASSERT`s that are sprinkled across the compiler. Arguably, they are all potential bug reports and that we should consider removing them, but they are not that evil and makes the code more self-documenting and prevents more serious issues. I could make it an `INT_ASSERT` if you think that'd make the code less scary.

I could always take your initial suggestion as well. But I'd prefer to make it a separate effort (ideally after 2.2).

Reply:
That's alright, I just wanted to see if there was an obvious way to avoid an assertion.